test/models/settings/email_setting_test.rb

```diff
@@ -0,0 +1,53 @@
+require 'test_helper'
+
+class EmailSettingTest < ActiveSupport::TestCase
+  def setup
+    Setting.cache.clear
+  end
+
+  context "when email.yaml exists" do
+    setup do
+      @delivery_settings = {"delivery_method" => :smtp,
+                           "smtp_settings" =>
+                             {"address" => "smtp.example",
+                             "authentication"=>"plain"},
+                           "sendmail_settings" =>
+                             {"arguments" => "args1"}}
+
+      Setting::Email.stubs(:mailconfig).returns(@delivery_settings)
+    end
+
+    test 'should use smtp settings' do
+      Setting::Email.create(:name => :delivery_method, :value => :smtp, :category => "Setting::Email")
+      Setting::Email.create(:name => :smtp_address, :value => "sm@example.com", :category => "Setting::Email")
+      Setting::Email.create(:name => :smtp_authentication, :value => "none", :category => "Setting::Email")
```

Comment:
Are these lines doing anything? Should they instead be `Setting[:smtp_authentication] = 'none'` etc?

I'm confused because the assertion on the line below checks if the delivery_settings equal the ones defined in the hash above, but not that it contains any of these modifications you're attempting to make on lines 21-23.